app/src/main/java/org/thoughtcrime/securesms/ui/components/SessionOutlinedButton.kt

```diff
@@ -0,0 +1,204 @@
+package org.thoughtcrime.securesms.ui.components
+
+import androidx.compose.animation.AnimatedVisibility
+import androidx.compose.animation.fadeIn
+import androidx.compose.animation.fadeOut
+import androidx.compose.foundation.BorderStroke
+import androidx.compose.foundation.interaction.MutableInteractionSource
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.RowScope
+import androidx.compose.foundation.layout.heightIn
+import androidx.compose.material.ButtonColors
+import androidx.compose.material.ButtonDefaults
+import androidx.compose.material.ButtonElevation
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.Shape
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.collectLatest
+import network.loki.messenger.R
+import org.thoughtcrime.securesms.ui.LaunchedEffectAsync
+import org.thoughtcrime.securesms.ui.LocalPalette
+import org.thoughtcrime.securesms.ui.contentDescription
+import kotlin.time.Duration
+import kotlin.time.Duration.Companion.seconds
+
+interface ButtonType {
```

Comment:
The other interface has its own class but there's this one here.
Can it be consolidated?
Between this class and the Button.kt?

Reply:
its own file, yeah, not 100% sure on if things should have their own files, I think I can also put `TextButton` in here... `ButtonType.Text` 🤔

Reply:
As long as we have a coherent logic. Currently we have ButtonSize in its own class and ButtonType declared in the Buttons.kt. 
We need consistency.
A lot of the things in the Compose files seems arbitrarily scaterred.

Reply:
@bemusementpark

Reply:
own file. done.